tests/admin_changelist/tests.py

```diff
@@ -1741,6 +1741,46 @@ def test_modifier_allows_multiple_section(self):
             self.assertIs(c.get_property("checked"), True)
         self.assertIs(checkboxes[-1].get_property("checked"), False)
 
+    def test_selection_counter_is_synced_when_page_is_show(self):
+        from selenium.webdriver.common.by import By
+
+        self.admin_login(username="super", password="secret")
+        self.selenium.get(self.live_server_url + reverse("admin:auth_user_changelist"))
+
+        form_id = "#changelist-form"
+        first_row_selector = "%s #result_list tbody tr:first-child" % form_id
+
+        selection_indicator = self.selenium.find_element(
+            By.CSS_SELECTOR, "%s .action-counter" % form_id
+        )
+        row_checkbox = self.selenium.find_element(
+            By.CSS_SELECTOR,
+            "%s .action-select" % first_row_selector,
+        )
+
+        # Select a row
+        row_checkbox.click()
+        self.assertEqual(selection_indicator.text, "1 of 1 selected")
+
+        # Go to another page and get back
+        detail_page_link = self.selenium.find_element(
+            By.CSS_SELECTOR,
+            "%s a" % first_row_selector,
+        )
+        detail_page_link.click()
```

Comment:
```suggestion
        self.selenium.get(self.live_server_url + reverse("admin:auth_group_changelist"))
```
Optional/Idea - might be enough just to navigate to a different page without finding a link and clicking